.github/workflows/tag-app-release.yml

```diff
@@ -0,0 +1,123 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate `app` input
+        shell: python
+        run: |
+          if "${{ inputs.app }}" not in ["api", "ingestion_server", "frontend"]:
+            raise ValueError("`app` input must be `api`, `ingestion_server`, or `frontend`. Received `${{ inputs.app}}`.")
+
+      - name: Validate `sha-tag` input
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let exists = undefined,
+              page = 0
+            while (!exists) {
+              page += 1
+              const { data: versions } =
+                await github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
+                  package_type: 'container',
+                  package_name: 'openverse-${{ inputs.app }}',
+                  org: 'WordPress',
+                  page,
+                  // max of `per_page`
+                  per_page: 100,
+                })
+              if (!versions.length) {
+                break
+              }
+              exists = versions.some((v) => v.metadata.container.tags.includes('${{ inputs.sha-tag }}'))
+            }
+            if (!exists) {
+              throw new Error(
+                `'${{ inputs.sha-tag }}' does not appear to be a valid SHA tag for ${{ inputs.app }}.`
+              )
+            }
+
+      - name: Calculate tag name
+        id: tag
+        run: |
+          # Format example: 2023.03.22.04.56.29
+          # `-u` forces UTC
+          formatted_date="$(date -u +%Y.%m.%d.%H.%M.%S)"
+          echo "date=$formatted_date" >> "$GITHUB_OUTPUT"
+          echo "tag=rel-$formated_date" >> "$GITHUB_OUTPUT"
+
+      - name: Add git tag
+        run: |
+          git tag -a ${{ steps.tag.outputs.tag }} ${{ inputs.sha-tag }}
+          git push origin ${{ steps.tag.outputs.tag }}
+
+      - name: Log in to GitHub Docker Registry
+        uses: docker/login-action@v2
+        with:
+          registry: https://ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Add new tag to existing docker image
+        run: |
+          docker buildx imagetools create ghrc.io/wordpress/openverse-${{ inputs.app }}:${{ inputs.sha-tag }} --tag ghrc.io/wordpress/openverse-${{ inputs.app }}:${{ steps.tag.outputs.tag }}
+
+      - name: Update draft release
+        # We don't actually ever use these releases, but because there's no easier
+        # way to generate the release notes for an application than this action,
+        # and because the action doesn't support generating the notes without
+        # adding them to a release (draft or no), we still let the release get created
+        # and then just copy the body from the result of this action.
+        uses: release-drafter/release-drafter@v5
+        id: release-drafter
+        with:
+          config-name: release_drafter_${{ inputs.app }}.yml
+          version: ${{ steps.tag.outputs.tag }}
```

Comment:
One concern with this - what does this diff _against_ when it's drafting the release? If it is naively only the last release that was made, we either get in a situation where:
1. Every subsequent changelog includes all previous changes since this workflow was introduced _plus_ the new changes
2. Every changelog only includes the changes _made to the `app` in question since whenever this workflow was last run_, regardless of `app` (e.g. if the runs were `api` -> `frontend` -> `api`, the latter `api` changelog would only have changes from `frontend` -> `api`.

Reply:
Oh jeez, that's a glaring issue with using release-drafter in this way. I believe it would just compare to the previous GitHub release. I think specifically we'd end up with the first case, each changelog including all previous changes for the app in subsequent changelogs.

I think the only way to get around this is going to be to generate the change list ourselves or find another action that allows you to define which commit to start from.

We could also fork release-drafter to add that capability to it ourselves (and while we're add it, the ability to stop it from creating the draft release and instead just build the markdown output).

I'll dig around and see if there's another action we can use or if we'll need to take an independent approach here.

Reply:
Digging more into it, other people use release-drafter with monorepos, so we should be able to do so as well: https://github.com/release-drafter/release-drafter/issues/1294

It looks like `tag-prefix` might tell release drafter how to find the "previous tag" to use when comparing. The docs aren't clear enough on it though, so I'll need to dig and find if that's the case.

Reply:
Yup, after looking at the code it does just that: when it searches for the "latestRelease" it filters the releases by `tagPrefix`: https://github.com/release-drafter/release-drafter/blob/master/lib/releases.js#L57-L59

_However_, this requires you to actually be using GitHub releases.

Which means we have a few options:

1. Go our own way and don't use release-drafter or use a fork of it modified to look at tags generally, not just GitHub releases
  - Only necessary if we're against GitHub release entirely
  - Big downside of needing a lot of extra work
2. Use release drafter the way guardrail is doing so and have it draft releases for each of our three applications. Create a new workflow triggered by releases that parses the release's name to get the name of the app and tags the image and then triggers the production deployment only for that app (by dispatching the corresponding production deployment workflow)
  - Benefit here is that it's pretty simple
  - One downside is that it will still be stuck using the meaningless semantic versioning. We could update the version template to just increment the major version number every time and drop the minor and patch number: https://github.com/release-drafter/release-drafter#projects-that-dont-use-semantic-versioning
  - Another downside is that we wouldn't have the changelogs on the documentation site, only in GitHub releases, in one, singular list (not split by app other than the prefix)
3. A hybrid of the two and the current approach: Stop drafting releases on an ongoing basis and instead use this workflow to create the release and immediately publish it. Automatically dispatch the production deployment with the tag generated. We would rename the workflow to just "Release" and keep the inputs as they are now.
  - Benefits:
    - Changelogs get added to the documentation site and can be split by application (making it easier to see the progression and cadence for individual apps in a list)
    - Lets us use CalVer instead of SemVer which is a more meaningful way of marking our releases
  - The only downside is that it's slightly more work than the second option. However, almost all of that work has already been done in this PR, we just have to make a few tweaks and fixes.

My vote is to do the third one and because it's the one that requires the medium amount of work of the three, I'll go ahead and update the branch do reflect it and then we can decide between it and the second option. To me the first option is a no-go, I just don't see why we would pour that much effort into this for so little benefit compared to the other two.

I'll make the following changes:

1. Remove the "draft release" workflow that currently creates the drafted releases on an ongoing basis for each push to main (this needs to happen anyway for this PR)
2. Change the workflow name to "Deploy production"
3. Change the app input to a choice, as you suggested in another comment
4. Update the release-drafter configuration to tell it to immediately publish the release
5. Dispatch the correct deployment workflow if the app was `api` or `frontend`.

Reply:
Thanks for doing all this investigation! I think your plan of action sounds great, and looks like we'd be prepending the app name anyway with this approach 💃🏼